decompiler/config/jak2/all-types.gc

```diff
@@ -24422,48 +24496,123 @@
 ;; water-h                        ;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-#|
+(defenum water-flags
+  :bitfield #t
+  :type uint32
+  (active)
+  (can-wade)
+  (can-swim)
+  (swim-ground)
+  (can-ground)
+  (use-ocean)
+  (tar)
+  (mud)
+  (deadly)
+  (use-water-anim)
+  (no-grab-ground)
+  (dark-eco)
+  (lava)
+  (swamp)
+  (flow)
+  (under-water)
+  (head-under-water)
+  (bouncing)
+  (wading)
+  (swimming)
+  (touch-water)
+  (jump-out)
+  (break-surface)
+  (spawn-drip)
+  (part-splash)
+  (part-drip)
+  (part-rings)
+  (part-water)
+  (event)
+  (over-water)
+  (find-water)
+  (touch-water-good)
+  )
+
 (deftype water-info (structure)
-  ()
+  ((trans       vector      :offset 0)
+   (normal      vector      :offset 16)
+   (base-height meters      :offset 32)
+   (depth       meters      :offset 36)
+   (handle      uint64      :offset 40)
+   (flags       water-flags :offset 48)
+   (prim        basic       :offset 52)
+   (extra-flags uint32      :offset 56)
+   )
```

Comment:
I think `trans` and `normal` here are probably `:inline` - with that I think these `:offset`s can become `:offset-assert`s